### DISCUSSION TITLE
continue

### DISCUSSION THREAD

**Discussion by yepyepyep4711** [djyt/cannonball#90] [Ideas]
Hi,

continuous mode currently changes the game's dynamics. You drive along all tracks sequentially. You can't choose which way you go, like in the original game. I wish there was the possibility to add a credit and start from the track where you lost, giving you the chance, with enough perseverance, to actually finish the game while maintaining your freedom of choice as to which way you want to go, which I feel is a big part of the appeal.

If this is possible, it would be great.

**Comment by djyt** [comment]
To be honest, I've never been completely content with the extra modes from a fun perspective. I consider them ok, but not great.
Maybe I will re-imagine them entirely at some point in the future.

**Reply by yepyepyep4711** [reply]
I'm afraid I have to agree here. What do you say about the idea of a continue feature like in many arcade games, when you can keep going as long as you still have credits ? I the case of OutRun I think my preferred variant would be "restart at the beginning of the current track".

**Reply by djyt** [reply]
You mean for the main game mode or the extra modes? I'm not that fussed about it for either right now. I mean, there's nothing to stop you from disabling the timer currently if you want to cheat....

**Reply by yepyepyep4711** [reply]
I mean for the main mode. Similar to options like "time: easy" under game engine.
I believe it's not the same than deactivating the timer altogether, fun-wise. If you deactivate the timer, this constraint just disappears. That's less fun, not more. If the timer is still there, but you have the possibility to start again at the beginning of your current track, then you still have that "damn, not quite there yet" experience when you run out of time, but without the somewhat bitter touch of "but I was nearly there and now I have to start from the very beginning". 
Especially if the restart is at the beginning of the track, with a fresh timer, then I believe it will make people play more. Because they know that right now, they have made it to that track. They just haven't beaten it yet. And if they don't add another credit and try again *right now*, then next time they will have to start from the top. It's really like the continue feature in, say, Street Fighter II: you've lost, but if you add a credit you'll get another shot at beating the game at the same stage.

Do you see what I mean?

I play the game in normal mode only. And I know that I always stop playing at the same point: when I don't want to race through the beginning *yet again* just to have another crack at the track where I lost.

**Reply by djyt** [reply]
I see what you mean, but it's not something I'd want to implement in the main codebase. OutRun is a 5 minute game, and quite an easy one at that. I don't think it needs a continue. You cab already practice levels in the time trial mode. 

There's nothing to stop you grabbing a copy of the code and changing things to your heart's content though - that's what the source code is for!

**Reply by yepyepyep4711** [reply]
> I see what you mean, but it's not something I'd want to implement in the main codebase. OutRun is a 5 minute game, and quite an easy one at that. 

I've never beaten it :'D
Neither now nor back then at the arcade.

> There's nothing to stop you grabbing a copy of the code and changing things to your heart's content though - that's what the source code is for!

If only I could. Sadly, that's out of my skills.
